dagshub/auth/oauth.py

```diff
@@ -1,57 +1,52 @@
-import getpass
-import sys
+import hashlib
 from typing import Optional, Dict
 import logging
-
-import pytimedinput
 import urllib
 import uuid
-from dagshub.common import config
-from dagshub.common.helpers import http_request
 
-CODE_INPUT_TIMEOUT = 60
+import httpx
+from dagshub.common import config
 
 logger = logging.getLogger(__name__)
 
 
 def oauth_flow(
     host: str,
-    client_id: Optional[str] = None,
-    code_input_timeout: Optional[int] = None,
+    client_id: Optional[str] = None
 ) -> Dict:
-    client_id = client_id or config.client_id
-    if code_input_timeout is None:
-        code_input_timeout = CODE_INPUT_TIMEOUT
-    state = uuid.uuid4()
+
     host = host.strip("/")
     dagshub_url = urllib.parse.urljoin(host, "login/oauth")
-    link_prompt = f"Go to {dagshub_url}/authorize?state={state}&client_id={client_id} and paste the code back in here."
-    code_prompt = "Code:"
-    if code_input_timeout <= 0:
-        print(link_prompt)
-        code = getpass.getpass(code_prompt)
-    else:
-        if not sys.__stdin__.isatty():
-            raise OauthNonInteractiveShellException(
-                "Can't perform OAuth in a non-interactive shell. "
-                "Please get a token using this command in a shell: dagshub login"
-            )
```

Comment:
You still need this check. If the user has no token, and the shell isn't interactive, it should quit right way.